src/vm.rs

```diff
@@ -419,6 +424,60 @@ fn matches_literal(s: &str, ix: usize, end: usize, literal: &str) -> bool {
     end <= s.len() && &s.as_bytes()[ix..end] == literal.as_bytes()
 }
 
+#[inline]
+fn matches_literal_casei(s: &str, ix: usize, end: usize, literal: &str) -> bool {
+    if end > s.len() {
+        return false;
+    }
+    if matches_literal(s, ix, end, literal) {
+        return true;
+    }
+    if s.is_char_boundary(ix) && s.is_char_boundary(end) && s.is_ascii() {
+        return s[ix..end].eq_ignore_ascii_case(literal);
+    }
+
+    // text captured and being backreferenced is not ascii, so we utilize regex-automata's case insensitive matching
+
+    let chars = literal.chars();
+
+    use alloc::boxed::Box;
```

Comment:
Is this `use` necessary?

Reply:
it was (for `--no-default-features`) but since I applied your suggestion it become unnecessary :tada:

Reply:
Oh I see :)